### PR TITLE
implement further type normalization logic

### PR DESCRIPTION
**Pull request by shaneburns**
account for variables of type bool and float in the normalizeTypes() method of the OverloadableMethodCadidate class - add/adjusts tests for these normalizations

### CODE REVIEW

**Comment by martn** [comment]
Oh please merge this one :+1:

File: src/OverloadedMethodCandidate.php [hunk 1]
```diff
@@ -86,7 +86,7 @@ private function mapArguments($reflectionFunction)
     private function normalizeTypes($types)
     {
         return array_map(function ($type) {
-            return $type == 'int' ? 'integer' : $type;
+            return $type == 'int' ? 'integer' : ($type == 'bool' ? 'boolean' : ($type == 'float' ? 'double' : $type));
```
Comment (mattstauffer) [inline src/OverloadedMethodCandidate.php:89]:
```suggestion
            $types = ['int' => 'integer', 'bool' => 'boolean', 'float' => 'double'];
            
            return array_key_exists($type, $types) ? $types[$type] : $type;
```

**Comment by shaneburns** [comment]
Thanks for the change request.  That's a much more elegant solution to the problem. I accepted the change, but just let me know if I need to take any further action.

**Comment by mattstauffer** [comment]
@shaneburns Nope, you're good. Thanks!

**Comment by mattstauffer** [comment]
https://github.com/tightenco/overload/releases/tag/v0.1.2

**Pull request merged**